doc/developer_guide/developer-guide.md

```diff
@@ -0,0 +1,100 @@
+# Notebook Connector Developer Guide
+
+## How to Release?
+
+Creating a new release of the NC requires 4 steps
+
+1. [Prepare the release](#prepare-the-release)
+2. Merge your pull request to the default branch
+3. [Switch to main branch and update it from remote](#switching-to-the-default-branch)
+3. [Create and push a tag to GitHub](#creating-a-git-tag-and-pushing-it-to-github)
+
+### Prepare the Release
+
+```shell
+poetry run nox -s release:prepare <additional-options> <version>
+```
+
+`<additional-options>` include
+* `--no-branch` if already working on a branch
+* `--no-pr` if you already created a pull request or plan to do so manually after preparing the release
+
+The command uses the nox task `project:fix` of the [Python Tool
+Box](https://github.com/exasol/python-toolbox/) (PTB) and
+* Optionally creates a branch
+* Optionally creates a pull request
+* Updates the [version number](#updating-the-version-number-without-preparing-a-release) and the Change Log
+
+Updating the Change Log includes
+* Moving file `doc/changes/unreleased.md` to a filename reflecting the specified version, e.g. `doc/changes/changes_1.2.3.md`
+* Updating the list of changes files in `doc/changes/changelog.md`
+* Adding the current date at the top of the changelog as the date of the release
+* Creating a new file `doc/changes/unreleased.md`
+
+### Switching to the default Branch
+
+You can find out the default branch with
+
+```shell
+git remote show origin | sed -n '/HEAD branch/s/.*: //p'
+```
```

Comment:
If this really necessary. Below we already assume that the main branch is `main`

Reply:
I can add the script to the command below, but I found it too complex and would prefer motivating users to name the default branch `main`.

If you prefer, then we can move the "find out default branch" after the other commands and add a sentence before "assuming default branch is main". Hm?